File: src/main/kotlin/inflearn/java/ch01_string/_1_9_숫자만_추출.java
package inflearn.java.ch01_string;

import java.util.Scanner;

public class _1_9_숫자만_추출 {
    public static Integer solution(String s) {
        String str = s.replaceAll("[^0-9]","");
        return Integer.parseInt(str);
    }

    public static Integer solution2(String s) {
        StringBuilder answer = new StringBuilder();
        for(char ch : s.toCharArray()) {
            if(Character.isDigit(ch)) answer.append(ch);
        }
        return Integer.parseInt(answer.toString());
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.next();

        System.out.println(solution(input));
    }
}

File: src/main/kotlin/inflearn/1_9 숫자만 추출.kt
package inflearn

import java.util.*

object `1_9 숫자만 추출` {
    fun solution(s: String): Int? {
        val str = s.replace("[^0-9]".toRegex(), "")
        return str.toInt()
    }

    fun solution2(s: String): Int? {
        val answer = StringBuilder()
        for (ch in s.toCharArray()) {
            if (Character.isDigit(ch)) answer.append(ch)
        }
        return answer.toString().toInt()
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val input = sc.next()
        println(solution(input))
    }
}